addons/website_sale/static/src/snippets/s_dynamic_snippet_products/options.js

```diff
@@ -28,8 +28,10 @@ const dynamicSnippetProductsOptions = s_dynamic_snippet_carousel_options.extend(
                 fields: ['id', 'res_id'],
             }
         }).then((data) => {
-            this.$target.get(0).dataset.filterId = data[0].res_id;
-            this.$target.get(0).dataset.numberOfRecords = this.dynamicFilters[data[0].res_id].limit;
+            if (data) {
```

Comment:
@JKE-be I think this is always truthy, even if you got no record, you should receive an empty array which is truthy, no?